src/libsyntax/ast.rs

```diff
@@ -1816,14 +1817,15 @@ impl Arg {
         }
     }
 
-    pub fn from_self(eself: ExplicitSelf, eself_ident: Ident) -> Arg {
+    pub fn from_self(attrs: Vec<Attribute>, eself: ExplicitSelf, eself_ident: Ident) -> Arg {
```

Comment:
Nit: could you use `ThinVec` here (since it's what the function actually wants).